templates/readme-template.md

```diff
@@ -0,0 +1,50 @@
+# Module Title
+
+## Description
+<!-- provide a summary of the purpose and use-case for your module -->
+
+## Requirements
+<!-- 
+list any kind of requirement or dependency for the usage of this module.
+ * components in a solution
+ * external dependencies
+ * other modules
+-->
+
+## Usage
+<!--
+ Provide an example configuration of the Module in a Solution.
+
+ Ensure all parameters are listed, with "/* optional */" where applicable
+-->
+```json
+{
+  "Modules" : {
+    "<module-id>" : {
+      "Provider" : "",
+      "Name" : "",
+      "Parameters" : {
+
+      }
+    }
+  }
+}
+```
+
+
+## Solution
+<!--
+  Provide a Solution-representation of the configuration that this module adds.
+  
+  Values that are dependant on Parameter Values should be identified as:
+  "<id-parameter-value>", "<other-parameter-value" etc.
+-->
+
+```json
+{
+	"Solution" : {
+      "Plugins" : {},
+      "Modules" : {}
+  }
+}
+```
```

Comment:
Not sure I follow this one. What would be an example here.

Reply:
This plays the role of the reference documentation for the module:

```json
{
    "Modules": {
        "githubauth": {
            "Provider": "githubidp",
            "Name": "cognito_github_api",
            "Parameters": {
                "id": {
                    "Key": "id",
                    "Value": "string"
                },
                "tier": {
                    "Key": "tier",
                    "Value": "string"
                },
                "githubClientId": {
                    "Key": "githubClientId",
                    "Value": "string"
                },
                "githubClientSecret": {
                    "Key": "githubClientSecret",
                    "Value": "string"
                },
                "githubOrg": {
                    "Key": "githubOrg",
                    "Value": "string"
                },
                "githubTeams": {
                    "Key": "githubTeams",
                    "Value": [
                        "string"
                    ]
                },
                "cognitoLink": {
                    "Key": "cognitoLink",
                    "Value": {
                        "Tier": "string",
                        "Component": "string",
                        "Instance": "string",
                        "Version": "string"
                    }
                }
            }
        }
    }
}
```